packages/pytest-simcore/src/pytest_simcore/hypothesis_type_strategies.py

```diff
@@ -1,9 +1,13 @@
 from hypothesis import provisional
 from hypothesis import strategies as st
-from pydantic import AnyHttpUrl, AnyUrl, HttpUrl
+from hypothesis.strategies import composite
+from pydantic import TypeAdapter
+from pydantic_core import Url
 
-# FIXME: For now it seems the pydantic hypothesis plugin does not provide strategies for these types.
-# therefore we currently provide it
-st.register_type_strategy(AnyUrl, provisional.urls())
-st.register_type_strategy(HttpUrl, provisional.urls())
-st.register_type_strategy(AnyHttpUrl, provisional.urls())
+
+@composite
+def url_strategy(draw):
+    return TypeAdapter(Url).validate_python(draw(provisional.urls()))
+
+
```

Comment:
so does it mean that works now like so?

Reply:
In the past Url was a `str`, so the provisional.urls() created the right types for the strategy. Now it's a `Url` and must be properly instantiated before "injecting it".